superset-frontend/src/dashboard/reducers/dashboardState.js

```diff
@@ -172,6 +173,12 @@ export default function dashboardStateReducer(state = {}, action) {
         fullSizeChartId: action.chartId,
       };
     },
+    [SET_DASHBOARD_REPORT]() {
+      return {
+        ...state,
+        report: action.report,
```

Comment:
do you need to put the report on the dashboard state if you have it also in the report state?